src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/DefaultLSPProgressListener.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.LanguageServer.Client;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Newtonsoft.Json.Linq;
+using System.Composition;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor.HtmlCSharp
+{
+    [Shared]
+    [Export(typeof(LSPProgressListener))]
+    internal class DefaultLSPProgressListener : LSPProgressListener, IDisposable
+    {
+        private readonly ILanguageServiceBroker2 _languageServiceBroker;
+
+        private ConcurrentDictionary<string, ProgressRequest> _activeRequests
+            = new ConcurrentDictionary<string, ProgressRequest>();
+
+        [ImportingConstructor]
+        public DefaultLSPProgressListener(ILanguageServiceBroker2 languageServiceBroker)
+        {
+            if (languageServiceBroker is null)
+            {
+                throw new ArgumentNullException(nameof(languageServiceBroker));
+            }
+
+            _languageServiceBroker = languageServiceBroker;
+
+            _languageServiceBroker.ClientNotifyAsync += ClientNotifyAsyncListenerAsync;
+        }
+
+        private Task ClientNotifyAsyncListenerAsync(object sender, LanguageClientNotifyEventArgs args)
+        {
+            return ProcessProgressNotificationAsync(args.MethodName, args.ParameterToken);
+        }
+
+        private async Task ProcessProgressNotificationAsync(string methodName, JToken parameterToken)
+        {
+            if (methodName != Methods.ProgressNotificationName ||
+               !parameterToken.HasValues ||
+               parameterToken["value"] is null ||
+               parameterToken["token"] is null)
+            {
+                return;
+            }
+
+            var token = parameterToken["token"].ToObject<string>(); // IProgress<object>>();
```

Comment:
Are `token` and `value` wellknown keys that are available somewhere?

Reply:
Yes/No

We can utilize `Methods.ProgressNotificationTokenName`, but nothing for value. Updated.